types/src/get_with_proof.rs

```diff
@@ -204,26 +203,7 @@ pub fn verify_update_to_latest_ledger_response(
         .map(|(req, res)| verify_response_item(ledger_info, req, res))
         .collect::<Result<Vec<_>>>()?;
 
-    // Verify ledger info signatures and potential epoch changes
-    if verifier.epoch_change_verification_required(ledger_info.epoch()) {
-        let epoch_change_li = validator_change_proof.verify(verifier)?;
-        let new_epoch_info = EpochInfo {
-            epoch: epoch_change_li.ledger_info().epoch() + 1,
-            verifier: Arc::new(
-                epoch_change_li
-                    .ledger_info()
-                    .next_validator_set()
-                    .ok_or_else(|| format_err!("No ValidatorSet in EpochProof"))?
-                    .into(),
-            ),
-        };
-        let new_verifier = VerifierType::TrustedVerifier(new_epoch_info.clone());
-        new_verifier.verify(ledger_info_with_sigs)?;
-        Ok(Some(new_epoch_info))
-    } else {
-        verifier.verify(ledger_info_with_sigs)?;
-        Ok(None)
-    }
+    trusted_state.verify_and_ratchet(ledger_info_with_sigs, validator_change_proof)
```

Comment:
nit: how about moving this up a little bit? (Meaning, if the server can't prove itself has a chain we can trust, why bother looking at the items?)